07_train_final_classifiers/scripts/run_all_genes.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+PARAMS_DIR=./02_classify_mutations/results/merged_all_params
+RESULTS_DIR=./07_train_final_classifiers/results/merged_all_params
+ERRORS_DIR=./final_merged_errors
+SEED=42
+
+mkdir -p $ERRORS_DIR
+
+# read the list of cosmic genes from tsv file
+merged_filename="data/cancer_genes/merged_with_annotations.tsv"
+
+read_genes_from_file() {
+    # create global gene array
+    declare -a -g genes
+
+    # read tab-separated file, genes should be the first column
+    while IFS=$'\t' read -r gene class; do
+        genes+=("$gene")
+    done < "$1"
+
+    # remove header
+    genes=("${genes[@]:1}")
+}
+read_genes_from_file $merged_filename
+
+for gene in "${genes[@]}"; do
+
+    # just do expression for now
+    # TODO: could do methylation but would need to pass PCA features
+    cmd="python 07_train_final_classifiers/train_classifier.py "
+    cmd+="--gene $gene "
```

Comment:
Cool, I didn't know you could do this (`+=`) in bash.

Reply:
Yeah! I think there are lots of ways to concatenate strings but this seems to work.